tests/verbose_output/test.sh

```diff
@@ -3,52 +3,41 @@
 set -u
 cd "$(dirname "$0")"
 
-## If verbose works
-
-# Launch distar in verbose mode and store output in a file
-../distar -v source_1.ml source_2.ml target.md > test.output
-
-# Keep diff output to compare in case of error
-output=$(diff test.output test.expected)
-
-# Check diff exit code
-if [ $? -eq 0 ]
-then
-    printf "\nGood output - OK"
-else
-    echo $output
-    exit 1
-fi
-
-
-## If verbose fails
-
-# Target doesn't exist
-../distar -v source_1.ml source_2.ml none.md 2> error.output
-
-output=$(diff error.output error.expected)
-
-# Check diff exit code
-if [ $? -eq 0 ]
-then
-    printf "\nBad target - OK"
-else
-    printf $output
-    exit 1
-fi
-
-# Source doesn't exist
-../distar -v none.ml target.md 2> source.output
-output=$(diff source.output source.expected)
-
-# Check diff exit code
-if [ $? -eq 0 ]
-then
-    printf "\nBad source - OK "
-else
-    printf -- "\n"$output
-    exit 1
-fi
+
+# Function to test if the verbose mode prints correct outputs
+verbose_mode_test () {
+    
+    # Launch distar in verbose mode and store output in a file
+    if [ $1 -eq 1 ] ; then
+        ../distar -v "$2" "$3" > "$4".output
+    else
+        ../distar -v "$2" "$3" 2> "$4".output
+    fi
+    
+    # Keep diff output to compare in case of error
+    output=$(diff "$4".output "$4".expected)
+
+    # If there is no difference between the files, diff return 0
+    # else it returns 1
+    if [ $? -eq 0 ];  then
+        printf "\nTest $4 - OK"    
+    else
+        echo "\n"$output
+        exit 1
+    fi
+}
+
+
+
+# Test with good arguments
+verbose_mode_test 1 "source_1.ml" "target.md" "good_args" 
```

Comment:
The captions of these tests are cryptic.

Reply:
Do I have to precise exactly why it works ?